assets/js/base/components/price-slider/index.js

```diff
@@ -0,0 +1,405 @@
+/**
+ * External dependencies
+ */
+import { sprintf, __ } from '@wordpress/i18n';
+import { Fragment, useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { useDebounce } from '@woocommerce/base-hooks';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { constrainRangeSliderValues, formatCurrencyForInput } from './utils';
+
+const PriceSlider = ( {
+	initialMin,
+	initialMax,
+	min,
+	max,
+	onChange,
+	step,
+	currencySymbol,
+	priceFormat,
+	showInputFields,
+	showFilterButton,
+	isLoading,
+} ) => {
+	const [ value, setValue ] = useState( {
+		min: initialMin,
+		max: initialMax,
+	} );
+	const [ formattedValue, setFormattedValue ] = useState( {
+		min: formatCurrencyForInput( value.min, priceFormat, currencySymbol ),
+		max: formatCurrencyForInput( value.max, priceFormat, currencySymbol ),
+	} );
+	const debouncedChangeValue = useDebounce( value, 500 );
+
+	useEffect( () => {
+		if (
+			value.min === undefined ||
+			value.max === undefined ||
+			min > value.min ||
+			max < value.max
+		) {
+			setValue( {
+				min,
+				max,
+			} );
+		}
+		setFormattedValue( {
+			min: formatCurrencyForInput(
+				value.min,
+				priceFormat,
+				currencySymbol
+			),
+			max: formatCurrencyForInput(
+				value.max,
+				priceFormat,
+				currencySymbol
+			),
+		} );
+	}, [ min, max ] );
+
+	useEffect( () => {
+		setFormattedValue( {
+			min: formatCurrencyForInput(
+				value.min,
+				priceFormat,
+				currencySymbol
+			),
+			max: formatCurrencyForInput(
+				value.max,
+				priceFormat,
+				currencySymbol
+			),
+		} );
+	}, [ value ] );
+
+	useEffect( () => {
+		if ( ! showFilterButton && ! isLoading ) {
+			triggerChangeCallback();
+		}
+	}, [ debouncedChangeValue ] );
+
+	const triggerChangeCallback = () => {
+		onChange( value );
+	};
```

Comment:
Since this is potentially passed through as a callback on the submit button if that's enabled, you need to wrap it in `useCallback`.  Dependencies would include `onChange`, `minPrice`, and `maxPrice`.  You'll also need to define this before the `useEffect` invocation right before it.